_episodes/05-loop.md

```diff
@@ -22,63 +22,45 @@ keypoints:
 - "Use `history` to display recent commands, and `!number` to repeat a command by number."
 ---
 
-**Loops** are key to productivity improvements through automation as they allow us to execute
-commands repeatedly. Similar to wildcards and tab completion, using loops also reduces the
-amount of typing (and typing mistakes).
+**Loops** are a programming construct which allow us to repeat a command or set of commands
+for each item in a list.
+As such they are key to productivity improvements through automation. 
+Similar to wildcards and tab completion, using loops also reduces the
+amount of typing required (and hence reduces the number of typing mistakes).
+
 Suppose we have several hundred genome data files named `basilisk.dat`, `unicorn.dat`, and so on.
-In this example,
+For this example,
 we'll use the `creatures` directory which only has two example files,
 but the principles can be applied to many many more files at once.
-We would like to modify these files, but also save a version of the original files, naming the copies
-`original-basilisk.dat` and `original-unicorn.dat`.
-We can't use:
-
-~~~
-$ cp *.dat original-*.dat
-~~~
-{: .language-bash}
-
-because that would expand to:
-
-~~~
-$ cp basilisk.dat unicorn.dat original-*.dat
-~~~
+We would like to print out the classification for each species, which is given on the second line of the file.
+For each file, we would need to execute the command `head -n 2` and pipe this to `tail -n 1`.
+We'll use a loop to solve this problem, but first let's look at the general form of a loop:
+
+```
+for thing in list_of_things
+do
+    operation_using $thing
+done
+```
 {: .language-bash}
 
-This wouldn't back up our files, instead we get an error:
-
-~~~
-cp: target `original-*.dat' is not a directory
-~~~
-{: .error}
+and we can apply this to our example like this:
 
-This problem arises when `cp` receives more than two inputs. When this happens, it
-expects the last input to be a directory where it can copy all the files it was passed.
-Since there is no directory named `original-*.dat` in the `creatures` directory we get an
-error.
-
-Instead, we can use a **loop**
-to do some operation once for each thing in a list.
-Here's a simple example that displays the first three lines of each file in turn:
-
-~~~
+```
 $ for filename in basilisk.dat unicorn.dat
 > do
->    head -n 3 $filename	# Indentation within the loop aids legibility
+>    head -n 2 $filename | tail -1	# Indentation within the loop is not required, but aids legibility
```

Comment:
Maybe this comment about indentation should be moved up to the new 'general form' code?

Reply:
Also, I wonder if we really need the "Indentation of code within a for loop" callout in addition to this comment.